test/unit/upload.js

```diff
@@ -46,60 +45,192 @@ describe('Upload Methods', function () {
       fs.readFile.restore()
     })
 
-    it('should fetch the repository an instance', function () {
-      return assert.isFulfilled(upload.uploadFile(mockArgs))
-        .then(function () {
-          sinon.assert.calledOnce(utils.getRepositoryAndInstance)
-          sinon.assert.calledWithExactly(
-            utils.getRepositoryAndInstance,
-            mockArgs
-          )
-        })
+    describe('Top level directory', function () {
```

Comment:
I actually dislike these tests. you're unit testing your in-file functions by triggering the entire file to run. maybe break some of these up after you move your in-file functions to module.exports to verify that (for example) the recursive caller makes multiple calls to create multiple folders, then all you have to do to test `.uploadFile` is that "the parent directory was asserted/created". does that make sense?